FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/external/samples/MecanumOdometrySample/FTCLibCommandSample/DriveSubsystem.java

```diff
@@ -0,0 +1,134 @@
+package org.firstinspires.ftc.robotcontroller.external.samples.MecanumOdometrySample.FTCLibCommandSample;
+
+import com.arcrobotics.ftclib.command.Subsystem;
+import com.arcrobotics.ftclib.drivebase.MecanumDrive;
+import com.arcrobotics.ftclib.gamepad.ButtonReader;
+import com.arcrobotics.ftclib.gamepad.GamepadEx;
+import com.arcrobotics.ftclib.gamepad.GamepadKeys;
+import com.arcrobotics.ftclib.geometry.Pose2d;
+import com.arcrobotics.ftclib.hardware.ExternalEncoder;
+import com.arcrobotics.ftclib.hardware.JSTEncoder;
+import com.arcrobotics.ftclib.hardware.RevIMU;
+import com.arcrobotics.ftclib.hardware.motors.SimpleMotorImpl;
+import com.arcrobotics.ftclib.kinematics.MecanumOdometry;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+
+public class DriveSubsystem implements Subsystem {
+
+
+    GamepadEx driverGamepad;
+    Telemetry telemetry;
+
+    MecanumOdometry odometry;
+
+    //Gyro
+    RevIMU gyro;
+
+    public SimpleMotorImpl backLeftMotor, frontLeftMotor, backRightMotor, frontRightMotor;
+    public MecanumDrive driveTrain;
+    public JSTEncoder horizontalEncoder, leftEncoder, rightEncoder;
+
+    final double trackWidth = 16; // Distance from center of left wheel to center of right wheel
+    final double WHEEL_DIAMETER = 4; // Inches
+    final int PULSES_PER_ROTATION = 280; // NEVEREST 40
+
+
+    ButtonReader slowDownButton;
+
+
+    public DriveSubsystem(GamepadEx driverGamepad, HardwareMap hw, Telemetry telemetry) {
+        this.driverGamepad = driverGamepad;
+        this.telemetry = telemetry;
+
+        odometry = new MecanumOdometry(trackWidth);
+
+        backLeftMotor = new SimpleMotorImpl(hw, telemetry,"backLeftMotor");
+        frontLeftMotor = new SimpleMotorImpl(hw, telemetry,"frontLeftMotor");
+        backRightMotor = new SimpleMotorImpl(hw, telemetry, "backRightMotor");
+        frontRightMotor = new SimpleMotorImpl(hw, telemetry, "frontRightMotor");
+
+        horizontalEncoder = new JSTEncoder(hw, "hEncoder");
+        horizontalEncoder.setDistancePerPulse((WHEEL_DIAMETER * Math.PI) / PULSES_PER_ROTATION);
+
+        leftEncoder = new JSTEncoder(hw, "lEncoder");
+        leftEncoder.setDistancePerPulse((WHEEL_DIAMETER * Math.PI) / PULSES_PER_ROTATION);
+
+        rightEncoder = new JSTEncoder(hw, "hEncoder");
+        rightEncoder.setDistancePerPulse((WHEEL_DIAMETER * Math.PI) / PULSES_PER_ROTATION);
+        rightEncoder.setInverted(true);
+
+        gyro = new RevIMU(hw);
+
+
+        backLeftMotor.setInverted(true);
```

Comment:
@codeNinjaDev You don't need to invert the motors. The mecanum drivebase does it for you.

Reply:
Does the mecanum drivebase also work when the motor is inverted? I think it is good practice (I might be wrong) to explicitly reverse the motors.